cmd/sonobuoy/app/common.go

```diff
@@ -19,3 +19,11 @@ func getSonobuoyClient(cfg *rest.Config) (*client.SonobuoyClient, error) {
 	}
 	return client.NewSonobuoyClient(cfg, skc)
 }
+
+func getSonobuoyClientFromKubecfg(kubecfg Kubeconfig) (*client.SonobuoyClient, error) {
```

Comment:
I could see there also being a mustGetSonobuoyClientFromKubecfg function that panics/exits on an error and doesn't return an error since this function generally is used with os.Exit(1).

Reply:
I had done that originally but it was counter to my eventual goal. Having looked at the codebase (and having done something like this before) I'd like to switch to `RunE` for the cobra commands and centralize all the error handling there.

That way as much as possible I can centralize the error reporting, os.Exit calls, and make each of the functions as testable as possible. Having the os.Exit calls throughout can make it difficult to test certain paths.